frontend/services/viewer/viewerService/viewerController.tsx

```diff
@@ -0,0 +1,666 @@
+				params.account,
+				params.model,
+				params.id ? [params.id] : params.ids
+		);
+	}
+
+	/**
+	 * Settings
+	 */
+
+	public updateViewerSettings(settings) {
+		if (settings) {
+			this.settings = settings;
+			if (this.settings.properties && this.settings.properties.unit) {
+				this.setUnits(this.settings.properties.unit);
+			}
+		}
+	}
+
+	public setUnits(units) {
+		this.units = units;
+
+		if (units === 'mm') {
+			this.convertToM = 0.001;
+		} else if (units === 'ft') {
+			this.convertToM = 0.0032;
+		}
+
+		// Set the units in unity for the measure tool
+		if (this.units) {
+			UnityUtil.setUnits(this.units);
+		}
+	}
+
+	/**
+	 * Nav mode
+	 */
+
+	public setNavigationMode(mode) {
+		this.setNavMode(mode, false);
+	}
+
+	public setNavMode(mode, force) {
+		if (this.currentNavMode !== mode || force) {
+			this.currentNavMode = mode;
+			this.setNavigation(mode);
+		}
+	}
+
+	public navMethodChanged(newNavMode) {
+		this.currentNavMode = newNavMode;
+	}
+
+	public get getNavMode() {
+		return this.currentNavMode;
+	}
+
+	/**
+	 * Fullscreen
+	 */
+
+	public switchFullScreen() {
+		if (!this.fullscreen) {
+			if (this.viewer.hasOwnProperty('mozRequestFullScreen')) {
+			} else if (this.viewer.webkitRequestFullscreen) {
+				this.viewer.webkitRequestFullscreen();
+			}
+
+			this.fullscreen = true;
+		} else {
+			if (document.webkitCancelFullScreen) {
+				document.webkitCancelFullScreen();
+			}
+
+			this.fullscreen = false;
+		}
+	}
+
+	/**
+	 * Pins
+	 */
+
+	public addPin = async ({account, model, id, type, pickedPos: position, norm, colours, viewpoint, isSelected}) => {
+			await this.isViewerReady();
+			this.pins[id] = new Pin({ id, type, position, norm, colors: colours, account, model });
+
+			if (isSelected) {
+				this.selectPin(id);
+			}
+			await this.isModelReady();
+			if (type === 'risk') {
+				await this.dropRiskPin(id, position, norm, colours);
+			} else {
+				await this.dropIssuePin(id, position, norm, colours);
+			}
+	}
+
+	public removePin = ({id}) => {
+		if (this.pins.hasOwnProperty(id)) {
+			this.pins[id].remove();
+			delete this.pins[id];
+		}
+	}
+
+	public removeUnsavedPin() {
+		this.removePin({ id: this.newPinId });
+		this.setPin({ data: null });
+	}
+
+	public highlightPin(id) {
+		// If a pin was previously highlighted switch it off
+		if (this.previousHighLightedPin) {
+			this.previousHighLightedPin.highlight();
+			this.previousHighLightedPin = null;
+		}
+
+		// If the pin exists switch it on
+		if (id && this.pins.hasOwnProperty(id)) {
+			this.pins[id].highlight();
+			this.previousHighLightedPin = this.pins[id];
+		}
+	}
+
+	public changePinColor({id, colours}) {
+		if (this.pins.hasOwnProperty(id)) {
+			this.pins[id].changeColour(colours);
+		}
+	}
+
+	public setPinDropMode(on: boolean) {
+		this.pinDropMode = on;
+
+		if (on) {
+			MultiSelect.toggleAreaSelect(false);
+		}
+	}
+
+	public setPin(data) {
+		this.pinData = data;
+	}
+
+	public getPinData(): any {
```

Comment:
I don't understand the purpose of this function.  (or the whole pinData variable). It doesn't sound like it needs to exist. a quick grep shows it's being used in risk comments, risk comments don't have pins so I don't know why it's passing through a position with the comment.

@sanmont3drepo  @Charence what's going on here?

Reply:
I think is being used when you edit/drop a pin. Im not sure is the best way to do it.